File: app/src/main/java/info/nightscout/androidaps/utils/SPBackupAgent.java
package info.nightscout.androidaps.utils;

import android.app.backup.BackupAgentHelper;
import android.app.backup.SharedPreferencesBackupHelper;

public class SPBackupAgent extends BackupAgentHelper {

    @Override
    public void onCreate() {
        // API 24
        final String PREFS = getApplicationContext().getPackageName() + "_preferences";
        final String PREFS_BACKUP_KEY = "SP";
        SharedPreferencesBackupHelper helper =
                new SharedPreferencesBackupHelper(this, PREFS);
        addHelper(PREFS_BACKUP_KEY, helper);
    }
}

File: app/src/main/kotlin/app/aaps/utils/SPBackupAgent.kt
package app.aaps.utils

import android.app.backup.BackupAgentHelper
import android.app.backup.SharedPreferencesBackupHelper
import app.aaps.shared.impl.sharedPreferences.preferencesFileName

@Suppress("LocalVariableName", "unused")
class SPBackupAgent : BackupAgentHelper() {

    override fun onCreate() {
        // API 24
        val PREFS = preferencesFileName(applicationContext)
        val PREFS_BACKUP_KEY = "SP"
        val helper = SharedPreferencesBackupHelper(this, PREFS)
        addHelper(PREFS_BACKUP_KEY, helper)
    }
}